src/gengowatcher/event_bus.py

```
@@ -0,0 +1,110 @@
+                # Peek at queue tail for recent duplicate events
+                for i in range(min(10, q.qsize())):
+                    try:
+                        existing = q.queue[-(i + 1)]  # type: ignore
+                        if (existing.get("type") == event.type and
+                            existing.get("collection_id") == event.collection_id):
+                            # Skip this consumer, but continue to others
+                            should_skip = True
+                            break
+                    except (IndexError, AttributeError, KeyError):
+                        continue
+            except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)